pkgs/top-level/all-packages.nix

```diff
@@ -10251,7 +10251,7 @@ with pkgs;
   silc_server = callPackage ../servers/silc-server { };
 
   sile = callPackage ../tools/typesetting/sile {
-    lua = lua5_4;
+    lua = lua5_3;
```

Comment:
Hmm, there should be a different fix. SILE only requires `compat53` on Lua 5.2 or older. It doesn't need it on 5.3 or above — nor does anything for that matter. That should be fixed in Nix packaging somewhere (note it is kind of silly that setting this to 5.3 works since the library in question is a ship for *less than* 5.3 to shim features.

Reply:
If I recall correctly the `compat53` rock loads fine in 5.3 and 5.4, it just is pretty close to a noop and not needed.

Reply:
I'm not sure if that should be fixed in this PR or not. Not sure what the culturally acceptable option is here.

Reply:
@alerque, Given that sile only needs this for 5.3 or lower is the better fix to just not add it to the dependencies in the sile package? I just tried and it builds just fine that way.

Reply:
Me either. But sile is working with Lua 5.4 on other platforms and we just went though a big discussion to get it there, so this feels like a needless regression.

Reply:
Never mind, I was wrong about my test. I still had 5.3 there. When I went back to 5.4 a missing compat53 breaks it.

Reply:
Hrmmm... Looks like this is the problematic line perhaps? https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/lua-modules/generated-packages.nix#L263

cc: @vcunat, I don't want to go messing with that line without understanding it's impact.

Reply:
I've reverted this change and now it seems to be working. Not sure why exactly but I think we can close this unless it crops up again for some reason.